web/app/js/components/ResourceDetail.jsx

```diff
@@ -21,6 +21,9 @@ import _reduce from 'lodash/reduce';
 import { processNeighborData } from './util/TapUtils.jsx';
 import { withContext } from './util/AppContext.jsx';
 
+// if there has been no traffic for some time, show a warning
+const showNoTrafficMsgDelay = 1000;
```

Comment:
I think it would be good to be more explicit about how long of a duration `showNoTrafficMsgDelay` is. What do you thinking about changing it to `showNoTrafficMsgDelayMs`?

Reply:
Also, should the duration be longer? What criteria should we use to pick the right duration? Is this variable used for debugging purposes to indicate if something is wrong or is it purely informative?

Reply:
whoooooops I changed it to 1 sec for testing and forgot to change it back. I think a 5-10 second delay would be more appropriate!

Reply:
basically I don't want it to be flickering if there's intermittent traffic (low RPS from the service) but I do want it to appear soon enough so that the user sees it before they leave the empty page in confusion

Reply:
Should we have an API for this? It would be nice to know that we got metrics at *some* point, even though they're not in the current window. Feels related-ish to #1875.

Reply:
yes! same idea. it would be nice to know if we got metrics at some point. we could add this into the metrics response somehow.... or maybe put this in the get api

Reply:
I agree -- seems like we could introduce an API that does a historical prometheus query for this information. Trying to determine based on current traffic seems difficult, and it can lead to the flickering issue that @rmars mentioned:



![screenrecording20190111at2](https://user-images.githubusercontent.com/9226/51062547-1ad16700-15ac-11e9-8986-bf23307a166b.gif)